### PR TITLE
Support SciPy 1.13 and 1.14

### PR DESCRIPTION
**Pull request by asi1024**
Close https://github.com/cupy/cupy/issues/8272.

Blocked by:
- https://github.com/cupy/cupy/pull/8508
- https://github.com/cupy/cupy/pull/8509
- https://github.com/cupy/cupy/pull/8510
- https://github.com/cupy/cupy/pull/8526
- https://github.com/cupy/cupy/pull/8527
- https://github.com/cupy/cupy/pull/8532
- https://github.com/cupy/cupy/pull/8533
- https://github.com/cupy/cupy/pull/8546
- https://github.com/cupy/cupy/pull/8547

https://ci.preferred.jp/cupy.linux.cuda125/170358/

### CODE REVIEW

**Comment by kmaehashi** [comment]
Could you bump the version in `docs/requirements.txt` as well?

File: .pfnci/schema.yaml [hunk 1]
```diff
@@ -291,6 +291,19 @@ numpy:
...
       - "3.11"
       - "3.12"
+  # "2.0":
+  #   spec: "==2.0.*"
+  #   python:
+  #     - "3.10"
+  #     - "3.11"
+  #     - "3.12"
+  # "2.1":
+  #   spec: "==2.0.*"
```
Comment (kmaehashi) [inline .pfnci/schema.yaml:301]:
```suggestion
  #   spec: "==2.1.*"
```

**Comment by asi1024** [comment]
There has been an unidentified issue leading to the failure of cuda-python tests with SciPy 1.14. As a temporary workaround in this pull request, we use an older version of SciPy to pass CI. The root cause of the problem will be addressed in a separated issue.

**Comment by asi1024** [comment]
@kmaehashi PTAL!

**Comment by kmaehashi** [comment]
Thanks, LGTM!

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**